src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -721,4 +727,105 @@ export class WorkspacesUsecases {
   }> {
     return this.teamRepository.getTeamUserAndTeamByTeamId(userUuid, teamId);
   }
+
+  async deleteWorkspaceContent(
+    workspaceId: Workspace['id'],
+    user: User,
+  ): Promise<void> {
+    const workspace = await this.workspaceRepository.findById(workspaceId);
+
+    if (!workspace) {
+      throw new NotFoundException('Workspace not found');
+    }
+
+    if (!workspace.isUserOwner(user)) {
+      throw new ForbiddenException('You are not the owner of this workspace');
+    }
+    const items =
```

Comment:
This is not going to scale. We have a way of handling this: mark the workspace root folder as deleted (`removed=true`). A background process will handle the entire tree deletion for you. No delay and simpler. 

For more info, check [this](https://inxt.atlassian.net/wiki/spaces/T/pages/121733121/DFF+-+v2)